### ISSUE TITLE
Missing Itemtype functions

### ISSUE THREAD

**Issue by Codinablack** [otland/forgottenserver#2293]
Ok so I can't do much in the world of c++ right now but I can do some things and one thing I would like to do is to have a couple more itemtype functions available for everyone on a stock tfs. 

Itemtype(..):getAmmoType()
Itemtype(..):isGroundTile()

There are a few others that may be nice to have but aren't really that necessary as there are other ways of finding this information such as

Itemtype(..):isBlocking()
Itemtype(..)isMagicField()
Itemtype(..)isUseable()
Itemtype(..)isHangable() -- _removed_
Itemtype(..)isRotateable() -- _removed_
Itemtype(..):isPickupable()
Itemtype(..):hasWalkStack() -- _removed_
ItemType(..):getCorpseType() -- **added**

I can make each one of these and put in a pull request to have it merged, however I figure most of these aren't already there as they were deemed not necessary. I strongly request the top two atleast and I ask here what of the others would be accepted if I put in the pull request?

**Comment by WibbenZ** [comment]
IMO they would all kinda be good to have, really annoying when you don't have a function that you need :p

**Comment by ranisalt** [comment]
Please do. Most of those functions are actually much similar to others, you might get away copying them and changing the function to call.

Also, I think as both are constant values you can push as attributes instead of methods.

**Comment by Codinablack** [comment]
I guess I will ultimately have to make some alterations because I believe a couple aren't working correctly, hasWalkStack(), isPickupable(), the two I notice right now, isUsuable(), getAmmoType(), isGroundTile(), isMagicField(), isBlocking(), are all for sure working fine as far as I can tell. Still more testing, if anyone see's my errors please go ahead and point them out, I'll fix them when I get a chance tomorrow or later, It is late now I am going to bed

**Comment by thexamx** [comment]
@Codinablack 
About pickupable did you test like this:
to items.h
```
bool isPickupable() const {
	return (pickupable);
}
```
`pushBoolean(L, itemType->isPickupable());`

**Comment by Codinablack** [comment]
@thexamx Thank you for that, I will definitely use it 👍

**Comment by Codinablack** [comment]
The isHangable() method has been removed from the list due to isHangable already being a bool value for itemType, and would require either renaming that bool in many places in source, or figuring a better name for the method. As is, I feel that would be better made in lua anyways. 

The isRotateable() method as been removed from the list for all the same reasons as isHangable()

The hasWalkStack() method has been removed because it always read true for everything, and couldn't figure out how to make it work the way I would have wanted/expected.

I added getCorpseType() to the list, as it was easy to do, and relatively useful. The return value is equivalent to the race type of the item.

There are others that could be added very easily that would definitely add much useful information pulled about the itemType for lua, but also alot could very well be useless because there are better ways through the item class. I will leave the list with just the alterations named above, because those I feel are the most useful, without being better or even possibly done in lua.
